aws_lambda_powertools/metrics/types.py

```diff
@@ -1,7 +1,51 @@
+from enum import Enum
+from typing import Dict
+
 from typing_extensions import NotRequired, TypedDict
 
 
+class MetricSummary(TypedDict):
+    Namespace: str
+    Metrics: Dict
+    Dimensions: Dict
+    Metadata: Dict
+
+
 class MetricNameUnitResolution(TypedDict):
     Name: str
     Unit: str
     StorageResolution: NotRequired[int]
+
+
+class MetricResolution(Enum):
+    Standard = 60
+    High = 1
+
+
+class MetricUnit(Enum):
+    Seconds = "Seconds"
+    Microseconds = "Microseconds"
+    Milliseconds = "Milliseconds"
+    Bytes = "Bytes"
+    Kilobytes = "Kilobytes"
+    Megabytes = "Megabytes"
+    Gigabytes = "Gigabytes"
+    Terabytes = "Terabytes"
+    Bits = "Bits"
+    Kilobits = "Kilobits"
+    Megabits = "Megabits"
+    Gigabits = "Gigabits"
+    Terabits = "Terabits"
+    Percent = "Percent"
+    Count = "Count"
+    BytesPerSecond = "Bytes/Second"
+    KilobytesPerSecond = "Kilobytes/Second"
+    MegabytesPerSecond = "Megabytes/Second"
+    GigabytesPerSecond = "Gigabytes/Second"
+    TerabytesPerSecond = "Terabytes/Second"
+    BitsPerSecond = "Bits/Second"
+    KilobitsPerSecond = "Kilobits/Second"
+    MegabitsPerSecond = "Megabits/Second"
+    GigabitsPerSecond = "Gigabits/Second"
+    TerabitsPerSecond = "Terabits/Second"
+    CountPerSecond = "Count/Second"
```

Comment:
Consider adding [Literal](https://peps.python.org/pep-0586/) to annotate this.

Example:
Seconds = Literal["Seconds"]